### DISCUSSION TITLE
Multiple questions from a novice

### DISCUSSION THREAD

**Discussion by Timmiej93** [rbaron/b-parasite#164] [Q&A]
Note: If you feel like you saw this before, this used to be an issue. I didn't notice that the discussions section was available on this repo when posting that, so this is a migration of that issue to here, which is a more suitable location.

<hr>

I stumbled upon this project somewhere, and it looks to be exactly what I need. I've done some custom PCB projects before (QuinLED for example), but with this there seems to be a lot less guidance. Although I know the basics about most things here, I don't know which is the right option. I also don't know anything about PCB assembly by the factory, so here's a few questions from my side. I'd be happy to convert it into a Wiki article afterwards to help others in the future.

### The very first questions
**How active is this project?**
It doesn't seem like there has been a lot of activity recently. Is that because the project has reached its final form and doesn't need any changes anymore, or has it been more or less abandoned?

**Software**
I see there's different software options available. It seems like the bottom three (blinky, soil_read_loop and input) are mainly debugging/development firmwares, and the BLE and Zigbee options are for production. Given my experience with BT and Zigbee in my current setup, I'd MUCH prefer Zigbee on these devices. However, it's not clear to me if the difference between Zigbee and BLE is purely in firmware, or if there's a hardware change too.
**Q:** Do the BLE and Zigbee versions require different hardware?
**Q:** Since I'm not planning to use BT, is there a (cheaper) alternative MCU available that doesn't have BT?

**Ordering**
[This article](https://github.com/rbaron/b-parasite/wiki/How-to-order:-PCB-fabrication-and-SMT-assembly) is probably a great reminder for the main developer, and enough for the experienced person, but it probably leaves most people with lots of questions. It does for me at least. Let's run through my questions while trying to use JLCPCB. Some of them I could answer myself, but are probably helpful for others.
**Q:** Which gerber files do you upload? **A:** All of them. Zip the entire folder as mentioned in step 1 and upload it.
**Q:** Are PCB settings like the amount of layers, PCB thickness etc. all set correctly or is the default fine?
**Q:** Same question as above, but for assembly options.
**Q:** JCBPCB offers a "conformal coating" option. Would it ever be a good idea to use this, or should we just use something like Plastik 70?
**Q:** I'm ordering 50 units, which results in over 300 bucks worth of E73 modules and over 70 bucks of DFN T&H sensors. Are those prices acceptable or should I source parts elsewhere?
**Q:** A part is not available at JLCPCB, how do I chose a replacement? (In my specific case it's the MMBT3906)

_(Addition on 2023/10/22)_
**Q:** Can sensors safely be omitted? I notice that the light sensor and T&H sensor both have a relatively high parts price (€0,10 and €1,50 respectively, plus €3 for each sensor type since they're non-default components), while both are only nice to have. The light sensor is even useless when using a case, which also severely limits the usefulness of the T&H sensor.
**Q:** Same question about the 2 buttons. I see one button is a reset button, but I assume taking the battery out will have the same effect?

I'd love for somebody to make a video about this device, just so everything's clear, I don't understand how this is one of the few DIY solutions that hasn't gotten any attention yet, even though it seems to be the only decent wireless soil moisture meter out there.

**Comment by rbaron** [comment]
Hey @Timmiej93, I had some time this weekend and expanded my previous answers below. I will remove my previous answers so it's not confusing. Hopefully it's helpful to you and to other people who are new to the project.

> Q: Do the BLE and Zigbee versions require different hardware?
> Q: Since I'm not planning to use BT, is there a (cheaper) alternative MCU available that doesn't have BT?

All samples run on the same hardware, currently in revision 2.0.0. You can see what changed in each revision in the [Hardware Versions Wiki](https://github.com/rbaron/b-parasite/wiki/Hardware-Versions). For all revisions there is support for two pin-compatible modules, described [here](https://github.com/rbaron/b-parasite/wiki/Hardware-Versions#nrf-modules). Adding support for a different MCU will likely require hardware and software changes.

> Q: Are PCB settings like the amount of layers, PCB thickness etc. all set correctly or is the default fine?

Once you upload the gerbers zip file, the default values should work. People more familiar with the process may tweak them to their liking, but defaults should be fine.

> Q: Same question as above, but for assembly options.

The defaults should be fine, but sometimes you need to manually correct the orientation of some components, the light sensor being one of the usual ones, as described in [here](https://github.com/rbaron/b-parasite/wiki/How-to-order:-PCB-fabrication-and-SMT-assembly#components-orientations). There is an interactive render on JLCPCB that lets you correct the orientations. I usually spend some time checking that the dots align on the parts and on the footprints, and the pads are aligned with the components. Most of the time an JLCPCB engineer will do a check and email me with corrections/questions to confirm as part of their process.

> Q: JCBPCB offers a "conformal coating" option. Would it ever be a good idea to use this, or should we just use something like Plastik 70?

I haven't tried that. We have the discussion issue https://github.com/rbaron/b-parasite/issues/79 with some users' experiences and tips for conformal coating. I myself use the acrylic Plastik 70 spray these days.

> Q: I'm ordering 50 units, which results in over 300 bucks worth of E73 modules and over 70 bucks of DFN T&H sensors. Are those prices acceptable or should I source parts elsewhere?

I would recommend starting with 5-10 boards, seeing if the sensor meets your [expectations](https://github.com/rbaron/b-parasite/wiki/What-to-Expect-from-the-Sensor) and checking that everything works okay.

> Q: A part is not available at JLCPCB, how do I chose a replacement? (In my specific case it's the MMBT3906)

You usually have the following options: source them separately, find an electrically equivalent part, redesign the circuit or wait until they're back in stock. The 3906 specifically is a popular, common component, and most [alternatives](https://jlcpcb.com/parts/componentSearch?searchTxt=MMBT3906) should work if the selected one is unavailable. I would still compare the datasheets of the two parts to make sure they're compatible out of an excess of caution.

> Q: Can sensors safely be omitted? I notice that the light sensor and T&H sensor both have a relatively high parts price (€0,10 and €1,50 respectively, plus €3 for each sensor type since they're non-default components), while both are only nice to have. The light sensor is even useless when using a case, which also severely limits the usefulness of the T&H sensor.

Yes, but depending on which sample you use you will need to tweak the code, otherwise the default behavior will be "there's an issue initializing the sensor". If there's interest please file an issue for that and I'll try to help out.

> Q: Same question about the 2 buttons. I see one button is a reset button, but I assume taking the battery out will have the same effect?

Indeed previous hardware revisions (see [Hardware Versions](https://github.com/rbaron/b-parasite/wiki/Hardware-Versions)) did not have any buttons. With the development of the Zigbee sample, if you want to experiment with that, it's nice to have them both to factory reset the board. In theory you can do the same with a full reflash, but it's not as convenient.

Lastly, come say hi in the [Community Gallery](https://github.com/rbaron/b-parasite/issues/19) once you have your sensors up and running.

**Comment by drspangle** [comment]
@rbaron I know this isn't the main thread topic but could you elaborate on
your experience with that conformal coating? Has it held up, and on which
hardware revisions? Any issues?

**Reply by rbaron** [reply]
@drspangle I still have some 1.0.0 boards running since the start of the project 2y ago, and a few of each revision up to the latest 2.0.0. I honestly did not expect the Plastik 70 to work so well, but lately I just apply a single coating and basically forget about it. It took me a while to learn how to properly let the coating dry evenly (I built [this simultaneously overengineered and a joke of a spinning drying jig](https://github.com/rbaron/b-parasite/issues/63), but I'm pretty sure you can achieve the same with your hands and patience. I'd say this evenly drying made a difference, and once it's set I still let it fully dry for 1-2 days before putting it in the ground.

I'd still like to learn about better alternatives though. This is an area I'm not familiar with, I was just lucky that the first thing I researched and tried worked for me. I experimented a bit with epoxy resin from an arts and crafts kit I had, but I found it messy, wasteful and my application just looked terrible.

**Comment by drspangle** [comment]
@rbaron Thanks for that. You might have found the best formula actually.
Even consulting with some experts on ruggedization, they were suggesting
redesigning the sensor because they weren't convinced conformal coating
would work. I think I'm sold on your approach.

**Reply by rbaron** [reply]
@drspangle Thanks. Please take my input with a grain of salt. I know next to nothing about conformal coating except for these simple experiments, and I wouldn't use this for anything remotely professional. For an indoor, non critical fun project like this, I think it's an okay option.

**Reply by ScottG489** [reply]
@rbaron Should pretty much any ["clear acrylic spray"](https://www.amazon.com/clear-acrylic-spray/s?k=clear+acrylic+spray) work as a conformal coating? I'm looking to purchase this, but wanted to make sure this is what I should be getting:

https://www.amazon.com/Krylon-K01303007-Acrylic-Crystal-11-Ounce/dp/B00023JE7K

I wasn't able to find Plastik 70 for purchase. So if you have any other preferred suggestion for something I could buy on amazon that would be great :)

**Reply by drspangle** [reply]
@ScottG489 I don't think you're going to have much luck with that spray. It goes on quite thick. This is more what you're looking for if you want a ready-made spray: https://www.amazon.com/dp/B07TXWN3V6/

I am currently experimenting with dipping. I made a special 3d-printed dipping jig which uses minimal liquid and I am intending to use it with polyurethane conformal coating. If it fails I will go back to using the spray as @rbaron suggests.

**Reply by ScottG489** [reply]
Appreciate it @drspangle! I'll get that one.

Also unrelated to coatings, could you happen to suggest an SWD programmer? The [one linked in the wiki](https://github.com/rbaron/b-parasite/wiki/How-to-Flash-the-Firmware-Samples#swd-programmers) seems to have tripled in price since documented and [is now $60](https://www.adafruit.com/product/3571) from where I've found it.

For some reason I assumed I'd be able to flash with something like this: https://www.amazon.com/gp/product/B06W2KKW64

**Reply by ScottG489** [reply]
Actually I think I was able to find other sources for cheaper. I assume either of these would all be compatible with the rest of the wiki directions on flashing?

- https://www.waveshare.com/arm-debugger.htm
- https://www.amazon.com/ST-LINK-circuit-debugger-programmer-XYGStudy/dp/B00PVJ8Q4C

**Reply by ScottG489** [reply]
Ended up going with the first one and I actually got it flashed! Only small hiccup was I needed to re-plug it back in after the initial flash.

Btw I like that there was a docker command for building, but I was also able to use docker to flash using the following command:
```
docker run --privileged --rm -v ${PWD}:/workdir/project nordicplayground/nrfconnect-sdk:main \
  nrfjprog -f nrf52 --program /workdir/project/code/nrf-connect/samples/ble/build/zephyr/zephyr.hex --sectorerase
```

Shall I add this to the docs? I don't actually know how to contribute to the wiki though and from a little googling it doesn't appear possible? Could you advise how I could contribute to the wiki docs @rbaron?

**Reply by rbaron** [reply]
Glad to hear that you got it working @ScottG489. IIRC, Docker on mac OS does not support USB passthrough last time I looked into it, so I couldn't get the flashing with Docker to work on my system. Are you on Linux/Windows?

**Reply by ScottG489** [reply]
I'm on Linux, yes. Can definitely see that not working on other platforms. Still could be valuable to document it with a mention that it only works for Linux systems? It was very convenient!

**Reply by rbaron** [reply]
Awesome, I added your method [to the wiki](https://github.com/rbaron/b-parasite/wiki/How-to-Flash-the-Firmware-Samples#nrfjprog-with-docker-linux-only). Tks!
